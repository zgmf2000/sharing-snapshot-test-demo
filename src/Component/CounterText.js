//@flow
import {Text, View} from 'react-native';
import React, {Component} from 'react';

type Props = {
    count: String,
}

//TODO: Try to change the styling aspect on negation and positive
export default class CounterText extends Component<Props> {
    selectedStyle = (count) => {
        if (count < 0) {
            return negationStyle;
        }
        return positiveStyle;
    };

    render() {
        const {count} = this.props;
        const composedStyle = {...baseStyle,...this.selectedStyle(count)};
        return (
            <View>
                <Text style={composedStyle}>
                    {count}
                </Text>
            </View>
        );
    }
};
const baseStyle =  {
    fontSize: 30,
};
const negationStyle = {
    color: 'red',
    fontWeight: 'bold'
};
const positiveStyle = {
    color: 'green'
};
