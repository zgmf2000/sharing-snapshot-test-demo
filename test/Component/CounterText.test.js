import renderer from 'react-test-renderer';
import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


import CounterText from '../../src/Component/CounterText';

configure({adapter: new Adapter()});

describe('#counterColor', () => {
    describe('snapshot', () => {
        it('should match negation value', () => {
            const wrapper = renderer.create(<CounterText count={-1}/>);
            expect(wrapper.toJSON()).toMatchSnapshot();
        });

        it('should match positive value', () => {
            const wrapper = renderer.create(<CounterText count={2}/>);
            expect(wrapper.toJSON()).toMatchSnapshot();
        });

        it('should turn the color from red to green if change state value', () => {
            const wrapper = shallow(<CounterText count={-2}/>);
            let actualColor = wrapper.find('Text').props().style;

            expect(actualColor).toMatchSnapshot();
            wrapper.setProps({
                count: 3
            });
            actualColor = wrapper.find('Text').props().style;
            expect(actualColor).toMatchSnapshot();
        });
    });

    describe('manual', () => {
        describe('color', () => {
            it('should green if have negation value', () => {
                const wrapper = shallow(<CounterText count={2}/>);
                const actualColor = wrapper.find('Text').props().style.color;
                expect(actualColor).toEqual('green');
            });

            it('should red if have negation value', () => {
                const wrapper = shallow(<CounterText count={-1}/>);
                const actualColor = wrapper.find('Text').props().style.color;
                expect(actualColor).toEqual('red');
            });

            it('should turn the color from red to green if changed state value', () => {
                const wrapper = shallow(<CounterText count={-1}/>);
                let actualColor = wrapper.find('Text').props().style.color;

                expect(actualColor).toEqual('red');

                wrapper.setProps({
                    count: 2
                });

                actualColor = wrapper.find('Text').props().style.color;
                expect(actualColor).toEqual('green');
            });
        });

        describe('fontWeight', () => {
            it('should normal if positive value', () => {
                const wrapper = shallow(<CounterText count={2}/>);
                const actualColor = wrapper.find('Text').props().style.fontWeight;
                expect(actualColor).toEqual('normal');
            });

            it('should bold if negation value', () => {
                const wrapper = shallow(<CounterText count={-1}/>);
                const actualColor = wrapper.find('Text').props().style.fontWeight;
                expect(actualColor).toEqual('bold');
            });

            it('should turn the fontWeight to normal if changed state value', () => {
                const wrapper = shallow(<CounterText count={-1}/>);
                let actualColor = wrapper.find('Text').props().style.fontWeight;
                expect(actualColor).toEqual('bold');

                wrapper.setProps({
                    count: 2
                });
                actualColor = wrapper.find('Text').props().style.fontWeight;
                expect(actualColor).toEqual('normal');
            });
        });
    });
});
