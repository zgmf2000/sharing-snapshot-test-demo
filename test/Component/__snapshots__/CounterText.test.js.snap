// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#counterColor snapshot should match negation value 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "red",
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    -1
  </Text>
</View>
`;

exports[`#counterColor snapshot should match positive value 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "green",
        "fontSize": 30,
        "fontWeight": "normal",
      }
    }
  >
    2
  </Text>
</View>
`;

exports[`#counterColor snapshot should turn the color from red to green if change state value 1`] = `
Object {
  "color": "red",
  "fontSize": 30,
  "fontWeight": "bold",
}
`;

exports[`#counterColor snapshot should turn the color from red to green if change state value 2`] = `
Object {
  "color": "green",
  "fontSize": 30,
  "fontWeight": "normal",
}
`;
